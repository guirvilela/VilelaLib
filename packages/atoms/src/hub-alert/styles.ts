import styled from 'styled-components';

import { HubAlertProps } from './types';

export const AlertInnerStyle = styled.div`
  display: flex;
  gap: ${({ theme }) => theme.spaces[2]};
  margin-top: -2px;
  width: 100%;
`;

export const AlertTitleStyle = styled.span`
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme }) => theme.colors.neutral[800]};
`;

export const AlertContentStyle = styled.span`
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme }) => theme.colors.neutralGrey[700]};
  font-weight: 400;
  width: 100%;
  text-overflow: ellipsis;
  overflow: hidden;
`;

export const AlertCloseStyle = styled.button`
  background-color: transparent;
  width: ${({ theme }) => theme.spaces[3]};
  height: ${({ theme }) => theme.spaces[3]};
  border: none;
  color: inherit;
  cursor: pointer;
`;

export const AlertStyle = styled.div<Omit<HubAlertProps, 'title'>>`
  display: flex;
  width: 100%;
  background-color: ${({ theme, color }) =>
    color ? theme.colors[color][100] : theme.colors.neutral[100]};
  color: ${({ theme, color }) =>
    color ? theme.colors[color][500] : theme.colors.neutral[500]};
  border: 1px solid
    ${({ theme, color }) =>
      color ? theme.colors[color][500] : theme.colors.neutral[500]};
  border-radius: ${({ theme, radius }) =>
    radius ? theme.spaces[radius] : '4px'};
  align-items: ${({ inline }) => (inline ? 'center' : 'start')};

  ${({ theme, pad }) => {
    if (typeof pad === 'string') {
      return `padding: ${theme.spaces[pad]};`;
    }
    if (Array.isArray(pad)) {
      return `padding: ${pad.reduce(
        (acc, x) => acc + `${theme.spaces[x]} `,
        ''
      )};`;
    }
    return `padding: ${theme.spaces[2]};`;
  }}

  gap: ${({ theme, gap }) => (gap ? theme.spaces[gap] : theme.spaces[2])};

  ${AlertInnerStyle} {
    flex-direction: ${({ inline }) => (inline ? 'row' : 'column')};
    align-items: ${({ inline }) => (inline ? 'center' : 'start')};
  }

  ${AlertTitleStyle} {
    font-weight: ${({ inline }) => (inline ? 400 : 600)};
    white-space: ${({ nowrap }) => (nowrap ? 'nowrap' : undefined)};
  }
`;
