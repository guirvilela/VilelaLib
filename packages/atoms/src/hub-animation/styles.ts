import styled from 'styled-components';

import { HubAnimationProps } from './types';

export const AnimationStyle = styled.div<HubAnimationProps & { ref }>`
  ${({ theme, size, width, height }) => {
    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[2]};
          height: ${theme.spaces[2]};
          min-width: ${theme.spaces[2]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
          min-width: ${theme.spaces[3]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
          min-width: ${theme.spaces[5]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[6]};
          height: ${theme.spaces[6]};
          min-width: ${theme.spaces[6]};
        `;
      default:
        return `
          width: ${width || theme.spaces[5]};
          height: ${height || theme.spaces[5]};
          min-width: ${width || theme.spaces[5]};
        `;
    }
  }}
`;
