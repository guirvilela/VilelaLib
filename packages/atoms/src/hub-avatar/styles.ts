import styled from 'styled-components';

import { HubAvatarProps } from './types';

export const AvatarImgStyle = styled.img`
  width: 100%;
  height: 100%;
  background-size: cover;
  border-radius: 50%;
`;

export const AvatarInitialsStyle = styled.span`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  border-radius: 50%;
`;

export const AvatarStyle = styled.div<HubAvatarProps>`
  ${({ theme, size }) => {
    if (!size) {
      return `
        width: ${theme.spaces[7]};
        height: ${theme.spaces[7]};
        min-width: ${theme.spaces[7]};
      `;
    }

    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
          min-width: ${theme.spaces[3]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
          min-width: ${theme.spaces[5]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[7]};
          height: ${theme.spaces[7]};
          min-width: ${theme.spaces[7]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[10]};
          height: ${theme.spaces[10]};
          min-width: ${theme.spaces[10]};
        `;
      default:
        return `
          width: ${size};
          height: ${size};
          min-width: ${size};
        `;
    }
  }}
  ${({ borderless, theme }) =>
    borderless
      ? ''
      : `
    padding: ${theme.spaces[1]};
    border: 1px solid ${theme.colors.neutral[400]};
  `};

  border-radius: 50%;

  ${AvatarInitialsStyle} {
    ${({ theme, typo }) => `
      font-family: ${theme.typography[typo || 'h4'].family};
      font-size: ${theme.typography[typo || 'h4'].size};
      line-height: ${theme.typography[typo || 'h4'].lineHeight};
      letter-spacing: ${theme.typography[typo || 'h4'].letterSpacing};
    `}

    background-color: ${({ theme, color, colorLevel }) =>
      color && colorLevel
        ? theme.colors[color][colorLevel]
        : color && theme.colors[color]
        ? theme.colors[color][500]
        : color
        ? color
        : undefined};

    color: ${({ theme, colorText, colorTextLevel }) =>
      colorText && colorTextLevel
        ? theme.colors[colorText][colorTextLevel]
        : colorText && theme.colors[colorText]
        ? theme.colors[colorText][500]
        : colorText
        ? colorText
        : theme.colors.neutral[800]};
  }
`;
