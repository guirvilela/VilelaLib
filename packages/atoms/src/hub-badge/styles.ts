import styled, { css } from 'styled-components';

import { HubBadgeProps } from './types';

const allStyle = css<HubBadgeProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  gap: ${({ theme }) => theme.spaces[2]};

  ${({ theme, size, width, height }) => {
    if (width || height) {
      return `
        width: ${width};
        height: ${height};
      `;
    }

    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[2]};
          height: ${theme.spaces[2]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[4]};
          height: ${theme.spaces[4]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
        `;
      default:
        return '';
    }
  }}

  background-color: ${({ theme, color, colorLevel }) =>
    color && colorLevel
      ? theme.colors[color][colorLevel]
      : color
      ? theme.colors[color][100]
      : theme.colors.neutral[200]};

  color: ${({ theme, color, textColor, textColorLevel }) =>
    textColor && textColorLevel
      ? theme.colors[textColor][textColorLevel]
      : textColor
      ? theme.colors[textColor][500]
      : color
      ? theme.colors[color][500]
      : theme.colors.neutral[500]};

  border-radius: ${({ theme, rounded, radius }) =>
    rounded ? '50%' : theme.spaces[radius || '1']};

  ${({ theme, pad }) => {
    if (!pad) return `padding: 2px 6px`;
    if (typeof pad === 'string') return `padding: ${theme.spaces[pad]}`;
    if (pad.length === 4) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]} ${
        theme.spaces[pad[2]]
      } ${theme.spaces[pad[3]]}`;
    }
    if (pad.length === 3) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]} ${
        theme.spaces[pad[2]]
      }`;
    }
    if (pad.length === 2) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]}`;
    }
    return `padding: ${theme.spaces[pad[0]]}`;
  }};

  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  font-weight: 600;
`;

const floatStyle = css<HubBadgeProps>`
  position: relative;

  &::after {
    ${allStyle}

    position: absolute;
    content: '${({ content }) => content}';
    border: 1px solid ${({ theme }) => theme.colors.white[500]};
    ${({ theme, position }) => {
      if (position) {
        if (Array.isArray(position)) {
          return `
            top: -${theme.spaces[position[0]]};
            right: -${theme.spaces[position[1]]};
          `;
        }
        return `
          top: -${theme.spaces[position]};
          right: -${theme.spaces[position]};
        `;
      }
      return `
        top: -10px;
        right: -10px;
      `;
    }}
    ${({ theme }) => `
      min-width: ${theme.spaces[4]};
      width: ${theme.spaces[4]};
      height: ${theme.spaces[4]};
    `}
    padding: 0;

    font-family: ${({ theme }) => theme.typography.body3.family};
    font-size: ${({ theme }) => theme.typography.body3.size};
    line-height: ${({ theme }) => theme.typography.body3.lineHeight};
    letter-spacing: ${({ theme }) => theme.typography.body3.letterSpacing};
  }
`;

export const BadgeStyle = styled.div<HubBadgeProps>`
  ${({ float, content }) => {
    if (float) {
      if (content) {
        return floatStyle;
      }
    } else {
      return allStyle;
    }
    return '';
  }}
`;
