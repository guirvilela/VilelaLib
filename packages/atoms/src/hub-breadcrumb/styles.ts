import styled from 'styled-components';

import { HubBreadcrumbProps, HubBreadcrumbsProps } from './types';

export const BreadcrumbsStyle = styled.div<HubBreadcrumbsProps>`
  display: flex;
  align-items: center;
  overflow: auto;
`;

export const BreadcrumbsPathStyle = styled.div`
  display: flex;
  align-items: center;
`;

export const BreadcrumbsIconStyle = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  color: ${({ theme }) => theme.colors.neutral[400]};
  margin: ${({ theme }) => `0 ${theme.spaces[2]}`};
`;

export const BreadcrumbStyle = styled.span<HubBreadcrumbProps>`
  display: flex;
  align-items: center;
  gap: ${({ theme }) => theme.spaces[2]};
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme, active }) =>
    active ? theme.colors.primary[500] : theme.colors.neutral[400]};
  transition: color 0.5s;
  font-weight: 600;
  white-space: ${({ nowrap }) => (nowrap ? 'nowrap' : 'pre-wrap')};

  ${({ width }) =>
    width
      ? `
    display: block;
    max-width: ${width};
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
  `
      : ''}

  &:hover {
    color: ${({ theme }) => theme.colors.primary[500]};
  }
`;
