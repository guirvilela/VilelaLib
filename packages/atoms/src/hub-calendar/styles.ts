import styled from 'styled-components';

import { HubCalendarItem, HubCalendarProps } from './types';

export const CalendarHeadStyle = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: ${({ theme }) => theme.spaces[3]};
  width: 100%;
`;

export const CalendarHeadTitleStyle = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  gap: ${({ theme }) => theme.spaces[2]};

  font-family: ${({ theme }) => theme.typography.h4.family};
  font-size: ${({ theme }) => theme.typography.h4.size};
  line-height: ${({ theme }) => theme.typography.h4.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.h4.letterSpacing};
  font-weight: 500;
`;

export const CalendarHeadLinkStyle = styled.button`
  background-color: transparent;
  color: ${({ theme }) => theme.colors.primary[200]};
  border: none;

  :hover {
    text-decoration: underline;
    cursor: pointer;
  }
`;

export const CalendarHeadNextStyle = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  background-color: ${({ theme }) => theme.colors.neutral[200]};
  border-radius: 50%;
  cursor: pointer;

  height: ${({ theme }) => theme.spaces[6]};
  width: ${({ theme }) => theme.spaces[6]};
`;

export const CalendarHeadPrevStyle = styled(CalendarHeadNextStyle)`
  transform: rotate(-180deg);
`;

export const CalendarDaysStyle = styled.div`
  display: grid;
  align-items: center;
  justify-content: center;
  justify-items: center;
  gap: ${({ theme }) => theme.spaces[2]};
  grid-template-columns: auto auto auto auto auto auto auto;
  width: 100%;
`;

export const CalendarMonthsStyle = styled.div`
  display: grid;
  grid-template-columns: 50% 50%;
  width: 100%;
`;

export const CalendarYearsStyle = styled.div`
  display: grid;
  grid-template-columns: 50% 50%;
  width: 100%;
`;

export const CalendarPeriodStyle = styled.span`
  display: flex;
  align-items: center;
  justify-content: center;

  font-family: ${({ theme }) => theme.typography.body3.family};
  font-size: ${({ theme }) => theme.typography.body3.size};
  line-height: ${({ theme }) => theme.typography.body3.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body3.letterSpacing};
  font-weight: 400;
`;

export const CalendarDayStyle = styled.button<HubCalendarItem>`
  display: flex;
  align-items: center;
  justify-content: center;

  color: ${({
    theme,
    selected,
    active,
    previous,
    next,
    disabled,
    today,
    event
  }) => {
    if (selected || active) {
      if (previous || next) {
        if (event && event.color) {
          return theme.colors[event.color][300];
        }
        return theme.colors.neutral[500];
      }
      return theme.colors.white[500];
    }
    if (previous || next) {
      if (event && event.color) {
        return theme.colors[event.color][200];
      }
      return theme.colors.neutral[300];
    }
    if (disabled) {
      if (event && event.color) {
        return theme.colors[event.color][400];
      }
      return theme.colors.neutral[300];
    }
    if (today || (event && event.color)) {
      if (event && event.color) {
        return theme.colors[event.color][500];
      }
      return theme.colors.info[600];
    }
    return theme.colors.neutral[800];
  }};
  background-color: ${({
    theme,
    selected,
    active,
    today,
    event,
    previous,
    next
  }) => {
    if (previous || next) {
      if (event && event.color) {
        return theme.colors[event.color][400];
      }
    }
    if (selected) {
      if (event && event.color) {
        return theme.colors[event.color][500];
      }
      return theme.colors.info[500];
    }
    if (active) {
      if (event && event.color) {
        return theme.colors[event.color][300];
      }
      return theme.colors.info[300];
    }
    if (today || (event && event.color)) {
      if (event && event.color) {
        return theme.colors[event.color][100];
      }
      return theme.colors.info[100];
    }
    return 'transparent';
  }};
  border-radius: 50%;
  border: none;
  ${({ disabled }) => (disabled ? 'cursor: not-allowed;' : `cursor: pointer;`)};
  transition: 0.5s ease background-color, 0.2s ease color;
`;

export const CalendarMonthYearStyle = styled.button<HubCalendarItem>`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: ${({ theme }) => `${theme.spaces[1]} ${theme.spaces[2]}`};
  background-color: transparent;
  color: ${({ theme, selected, active }) => {
    if (selected) {
      return theme.colors.info[500];
    }
    if (!active) {
      return theme.colors.neutral[300];
    }
    return theme.colors.neutral[800];
  }};
  border-radius: 50%;
  border: none;
  cursor: pointer;

  :hover {
    text-decoration: underline;
  }
`;

export const CalendarActionsStyle = styled.div`
  display: grid;
  align-items: center;
  justify-content: end;
  gap: ${({ theme }) => theme.spaces[2]};
  width: 100%;
  padding: ${({ theme }) => `0 ${theme.spaces[2]}`};
`;

export const CalendarTodayStyle = styled.button`
  border: none;
  background-color: transparent;
  color: ${({ theme, disabled }) => {
    if (disabled) {
      return theme.colors.neutral[300];
    }
    return theme.colors.primary[200];
  }};
  ${({ disabled }) => (disabled ? '' : `cursor: pointer;`)};
`;

export const CalendarStyle = styled.div<HubCalendarProps>`
  display: flex;
  align-items: center;
  flex-direction: column;
  gap: ${({ theme }) => theme.spaces[2]};
  width: ${({ width }) => width || '100%'};

  ${CalendarDayStyle} {
    ${({ theme, size }) => {
      switch (size) {
        case 'xs':
          return `width: ${theme.spaces[2]}; height: ${theme.spaces[2]};`;
        case 'sm':
          return `width: ${theme.spaces[3]}; height: ${theme.spaces[3]};`;
        case 'md':
          return `width: ${theme.spaces[5]}; height: ${theme.spaces[5]};`;
        case 'lg':
          return `width: ${theme.spaces[6]}; height: ${theme.spaces[6]};`;
        default:
          if (size) {
            return `width: ${size}; height: ${size};`;
          }
          return `width: ${theme.spaces[6]}; height: ${theme.spaces[6]};`;
      }
    }};
  }

  ${CalendarTodayStyle},
  ${CalendarDayStyle} {
    font-family: ${({ theme, typo }) => theme.typography[typo || 'h4'].family};
    font-size: ${({ theme, typo }) => theme.typography[typo || 'h4'].size};
    line-height: ${({ theme, typo }) =>
      theme.typography[typo || 'h4'].lineHeight};
    letter-spacing: ${({ theme, typo }) =>
      theme.typography[typo || 'h4'].letterSpacing};
    font-weight: ${({ typo }) =>
      typo ? (typo.includes('body') ? 600 : 400) : 400};
  }

  ${CalendarHeadTitleStyle},
  ${CalendarHeadLinkStyle} {
    font-family: ${({ theme, typo, typoHeader }) =>
      theme.typography[typoHeader || typo || 'h4'].family};
    font-size: ${({ theme, typo, typoHeader }) =>
      theme.typography[typoHeader || typo || 'h4'].size};
    line-height: ${({ theme, typo, typoHeader }) =>
      theme.typography[typoHeader || typo || 'h4'].lineHeight};
    letter-spacing: ${({ theme, typo, typoHeader }) =>
      theme.typography[typoHeader || typo || 'h4'].letterSpacing};
    font-weight: ${({ typo, typoHeader }) =>
      typoHeader || typo
        ? (typoHeader || typo)?.includes('body')
          ? 600
          : 400
        : 400};
  }
`;
