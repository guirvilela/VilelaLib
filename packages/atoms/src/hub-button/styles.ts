import { shadeColor } from '@hubert/shared';
import { animations } from '@hubert/styles';
import styled, { css } from 'styled-components';

import { HubButtonProps } from './types';

const ButtonSizesStyle = css<HubButtonProps>`
  width: ${({ theme, full, size, variant }) => {
    if (full) return '100%';
    if (variant === 'icon') {
      switch (size) {
        case 'xs':
          return theme.spaces[5];
        case 'sm':
          return theme.spaces[6];
        case 'md':
          return theme.spaces[7];
        case 'lg':
          return theme.spaces[8];
      }
    }
    return 'auto';
  }};
  height: ${({ theme, size, variant }) => {
    switch (size) {
      case 'xs':
        return theme.spaces[5];
      case 'sm':
        return theme.spaces[6];
      case 'md':
        return theme.spaces[7];
      case 'lg':
        return theme.spaces[8];
      default:
        if (size === '100%') {
          return size;
        }
        if (variant === 'icon' || variant === 'action') {
          return 'auto';
        }
        return theme.spaces[7];
    }
  }};
  padding: ${({ theme, variant, size, noPadding }) => {
    if (variant === 'icon' || variant === 'action' || noPadding) {
      return undefined;
    }
    switch (size) {
      case 'xs':
        return `${theme.spaces[1]} ${theme.spaces[2]}`;
      default:
        return `${theme.spaces[1]} ${theme.spaces[3]}`;
    }
  }};
`;

const ButtonStylesStyle = css<HubButtonProps>`
  ${({
    theme,
    variant,
    color,
    colorLevel,
    bgColor,
    bgColorLevel,
    borderColor,
    borderColorLevel
  }) => {
    const hash =
      color && colorLevel
        ? theme.colors[color][colorLevel]
        : color && theme.colors[color]
        ? theme.colors[color][500]
        : color
        ? color
        : theme.colors.primary[500];
    const bgHash =
      bgColor && bgColorLevel
        ? theme.colors[bgColor][bgColorLevel]
        : bgColor && theme.colors[bgColor]
        ? theme.colors[bgColor][500]
        : bgColor
        ? bgColor
        : 'transparent';
    const borderHash =
      borderColor && borderColorLevel
        ? theme.colors[borderColor][borderColorLevel]
        : borderColor && theme.colors[borderColor]
        ? theme.colors[borderColor][500]
        : borderColor
        ? borderColor
        : hash;

    switch (variant) {
      case 'outline':
        return `
          background-color: ${bgHash};
          color: ${hash};
          border: 1.5px solid ${borderHash};
        `;
      case 'ghost':
      case 'icon':
      case 'action':
        return `
          background-color: ${bgHash};
          color: ${hash};
        `;
      default:
        return `
          background-color: ${hash};
          color: ${theme.colors.white[500]};
        `;
    }
  }}
`;

const ButtonHoverStyle = css<HubButtonProps>`
  &:hover:not(:disabled) {
    ${({ theme, variant, color, bgHover }) => {
      const hash =
        color === 'white'
          ? theme.colors.neutralGrey[300]
          : color && theme.colors[color]
          ? shadeColor(theme.colors[color][500], -5)
          : color
          ? color
          : theme.colors.neutral[600];

      if (bgHover) {
        switch (variant) {
          case 'default':
            return `
              background-color: ${hash};
            `;
          default:
            return `
              background-color: ${theme.colors.neutralGrey[100]};
            `;
        }
      }

      switch (variant) {
        case 'outline':
          return `
            background-color: ${theme.colors.neutralGrey[100]};
          `;
        case 'ghost':
        case 'icon':
        case 'action':
          return `
            color: ${hash};
          `;
        default:
          return `
            background-color: ${hash};
          `;
      }
    }}
  }
`;

const ButtonActiveStyle = css<HubButtonProps>`
  &:not(:disabled) {
    ${({ theme, color, colorLevel }) => {
      const hash =
        color && colorLevel
          ? theme.colors[color][colorLevel]
          : color && theme.colors[color]
          ? theme.colors[color][500]
          : color
          ? color
          : theme.colors.primary[500];
      return `
        outline: 1px solid ${hash};
      `;
    }}
  }
`;

const ButtonDisabledStyle = css<HubButtonProps>`
  &:disabled {
    ${({ theme, variant }) => {
      switch (variant) {
        case 'outline':
          return `
            border-color: ${theme.colors.primary[100]};
            color: ${theme.colors.primary[100]};
          `;
        case 'ghost':
        case 'icon':
        case 'action':
          return `
            color: ${theme.colors.primary[100]};
          `;
        default:
          return `
            background-color: ${theme.colors.neutral[300]};
          `;
      }
    }}
  }
`;

const ButtonCommonStyle = css<HubButtonProps>`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  gap: ${({ theme }) => theme.spaces[2]};
  border-radius: ${({ theme, rounded, radius }) =>
    rounded ? '50%' : theme.spaces[radius || 1]};
  border: none;
  font-weight: 600;
  transition: background-color, outline 0.5s;
  cursor: ${({ cursor, dragTrigger }) =>
    dragTrigger ? 'grab' : cursor || 'pointer'};
  white-space: nowrap;
  text-decoration: none;
`;

export const ButtonLoadingStyle = styled.div`
  width: ${({ theme }) => theme.spaces[5]};
  height: ${({ theme }) => theme.spaces[5]};
  border-radius: 50%;
  animation: 1s linear infinite ${() => animations.spin},
    ${() => animations.fadein} 1s ease;
`;

const ButtonShared = css<HubButtonProps>`
  ${ButtonCommonStyle}
  ${ButtonSizesStyle}
  ${ButtonStylesStyle}
  ${ButtonDisabledStyle}
  ${({ noHover }) => (noHover ? '' : ButtonHoverStyle)}
  ${({ active }) => (active ? ButtonActiveStyle : '')}

  ${({ clickable, disabled }) => {
    if (clickable === false) {
      return `
        pointer-events: none !important;
        cursor: not-allowed !important;
      `;
    }
    if (disabled) {
      return `
        cursor: not-allowed !important;
      `;
    }
    return '';
  }}

  ${ButtonLoadingStyle} {
    ${({ theme, variant, color, colorLevel }) => {
      switch (variant) {
        case 'outline':
        case 'ghost':
        case 'icon':
        case 'action':
          return `
              border: 3px solid ${
                color && theme.colors[color]
                  ? theme.colors[color][100]
                  : color
                  ? color
                  : theme.colors.primary[100]
              };
              border-top: 3px solid ${
                color && colorLevel
                  ? theme.colors[color][colorLevel]
                  : color && theme.colors[color]
                  ? theme.colors[color][500]
                  : color
                  ? color
                  : theme.colors.primary[300]
              };
            `;
        default:
          return `
              border: 3px solid rgba(255, 255, 255, 0.5);
              border-top: 3px solid ${
                color && colorLevel
                  ? theme.colors[color][colorLevel]
                  : color && theme.colors[color]
                  ? theme.colors[color][500]
                  : color
                  ? color
                  : theme.colors.white[500]
              };
            `;
      }
    }}
  }
`;

export const ButtonStyle = styled.button<HubButtonProps>`
  ${ButtonShared}
`;

export const ButtonLinkStyle = styled.a<HubButtonProps>`
  ${ButtonShared}
`;
