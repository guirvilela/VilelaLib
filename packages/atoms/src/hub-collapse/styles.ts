import { animations } from '@hubert/styles';
import styled from 'styled-components';

import { HubCollapseProps } from './types';

export const CollapseHeader = styled.div<{ disabled?: boolean }>`
  display: flex;
  gap: ${({ theme }) => theme.spaces[1]};
`;

export const CollapseHeaderTitle = styled.span`
  font-size: ${({ theme }) => theme.typography.body1.size};
  font-weight: 600;
`;

export const CollapseHeaderTrigger = styled.button`
  display: flex;
  justify-content: center;
  border: none;
  background-color: transparent;
  color: ${({ theme, disabled }) =>
    disabled ? theme.colors.neutral[200] : theme.colors.neutral[300]};
  cursor: ${({ disabled }) => (disabled ? 'not-allowed' : 'pointer')};
`;

export const CollapseContent = styled.div`
  animation: 0.3s ease-in-out ${() => animations.fadein};
  width: 100%;
`;

export const CollapseStyle = styled.div<HubCollapseProps>`
  display: flex;
  flex-direction: column;
  gap: ${({ theme, gap }) => theme.spaces[gap || 3]};
  width: ${({ width }) => width || '100%'};

  ${({ align }) => (align ? `align-items: ${align};` : '')}

  ${CollapseHeader} {
    ${({ titleClickable, disabled }) =>
      titleClickable
        ? disabled
          ? 'cursor: not-allowed;'
          : 'cursor: pointer;'
        : ''}
    ${({ titleAlign }) => (titleAlign ? `align-items: ${titleAlign};` : '')}
    width: ${({ titleWidth }) => titleWidth || '100%'};

    ${({ theme, titlePad }) => {
      if (!titlePad) return '';
      if (typeof titlePad === 'string')
        return `padding: ${theme.spaces[titlePad]};`;
      if (titlePad.length === 4) {
        return `padding: ${theme.spaces[titlePad[0]]} ${
          theme.spaces[titlePad[1]]
        } ${theme.spaces[titlePad[2]]} ${theme.spaces[titlePad[3]]};`;
      }
      if (titlePad.length === 3) {
        return `padding: ${theme.spaces[titlePad[0]]} ${
          theme.spaces[titlePad[1]]
        } ${theme.spaces[titlePad[2]]};`;
      }
      if (titlePad.length === 2) {
        return `padding: ${theme.spaces[titlePad[0]]} ${
          theme.spaces[titlePad[1]]
        };`;
      }
      return `padding: ${theme.spaces[titlePad[0]]};`;
    }}

    ${({ theme, titleColor, titleColorLevel }) =>
      titleColor
        ? `background-color: ${
            titleColorLevel
              ? theme.colors[titleColor][titleColorLevel]
              : theme.colors[titleColor]
              ? theme.colors[titleColor][300]
              : titleColor
          };`
        : ''}

    ${({ theme, titleRadius }) =>
      titleRadius ? `border-radius: ${theme.spaces[titleRadius]};` : ''}
  }

  ${CollapseHeaderTitle} {
    ${({ nowrap }) => (nowrap ? 'white-space: nowrap;' : '')}
    width: ${({ titleWidth }) => (titleWidth === 'auto' ? 'auto' : '100%')};
  }

  ${CollapseHeaderTrigger} > svg {
    transition: 0.3s ease-in-out transform;
  }

  ${CollapseContent} {
    ${({ height }) =>
      height
        ? `
      max-height: ${height};
      overflow-y: auto;
    `
        : ''}
  }
`;
