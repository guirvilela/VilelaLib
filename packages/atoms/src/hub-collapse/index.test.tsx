import { render } from '@testing-library/react';
import React from 'react';

import { wrapse } from '@hubert/shared';
import { ThemeProvider } from '@hubert/styles';

import { HubCollapse } from '.';

describe('HubCollapse component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<HubCollapse />, {
      wrapper: wrapse(ThemeProvider)
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
