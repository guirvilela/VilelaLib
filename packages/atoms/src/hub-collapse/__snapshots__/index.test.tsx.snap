// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HubCollapse component matches snapshot 1`] = `
<DocumentFragment>
  .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
}

.c4 {
  font-size: 16px;
  font-weight: 600;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: none;
  background-color: transparent;
  color: #CCD5DD;
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 16px;
  width: 100%;
}

.c0 .c1 {
  width: 100%;
}

.c0 .c3 {
  width: 100%;
}

.c0 .c5 > svg {
  -webkit-transition: 0.3s ease-in-out -webkit-transform;
  -webkit-transition: 0.3s ease-in-out transform;
  transition: 0.3s ease-in-out transform;
}

<div
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <span
        class="c3 c4"
      />
      <button
        class="c5 c6"
        type="button"
      >
        <svg
          fill="none"
          height="24"
          style="transform: rotate(180deg);"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.9999 13.4108L12.7099 9.1708C12.617 9.07707 12.5064 9.00268 12.3845 8.95191C12.2627 8.90114 12.132 8.875 11.9999 8.875C11.8679 8.875 11.7372 8.90114 11.6154 8.95191C11.4935 9.00268 11.3829 9.07707 11.2899 9.1708L7.04995 13.4108C6.95622 13.5038 6.88183 13.6144 6.83106 13.7362C6.78029 13.8581 6.75415 13.9888 6.75415 14.1208C6.75415 14.2528 6.78029 14.3835 6.83106 14.5054C6.88183 14.6272 6.95622 14.7378 7.04995 14.8308C7.23731 15.017 7.49076 15.1216 7.75495 15.1216C8.01913 15.1216 8.27259 15.017 8.45995 14.8308L11.9999 11.2908L15.5399 14.8308C15.7262 15.0155 15.9776 15.1197 16.2399 15.1208C16.3716 15.1216 16.502 15.0963 16.6239 15.0466C16.7457 14.9968 16.8565 14.9235 16.9499 14.8308C17.047 14.7412 17.1254 14.6332 17.1805 14.5131C17.2356 14.3931 17.2664 14.2633 17.271 14.1312C17.2757 13.9992 17.2541 13.8675 17.2076 13.7439C17.161 13.6202 17.0905 13.507 16.9999 13.4108Z"
            fill="#CCD5DD"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
